wrangler-service/src/main/java/io/cdap/wrangler/service/schema/SchemaViewerHandler.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.wrangler.service.schema;
+
+import com.google.gson.JsonElement;
+import io.cdap.cdap.api.annotation.TransactionControl;
+import io.cdap.cdap.api.annotation.TransactionPolicy;
+import io.cdap.cdap.api.service.http.HttpServiceRequest;
+import io.cdap.cdap.api.service.http.HttpServiceResponder;
+import io.cdap.cdap.api.service.http.SystemHttpServiceContext;
+import io.cdap.cdap.spi.data.StructuredTableContext;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import io.cdap.wrangler.dataset.schema.HealthcarePackageSchemaLoader;
+import io.cdap.wrangler.dataset.schema.SchemaFile;
+import io.cdap.wrangler.dataset.schema.SchemaFile.Discriminator;
+import io.cdap.wrangler.dataset.schema.SchemaFileDescriptor;
+import io.cdap.wrangler.dataset.schema.SchemaFileStore;
+import io.cdap.wrangler.proto.NotFoundException;
+import io.cdap.wrangler.proto.ServiceResponse;
+import io.cdap.wrangler.service.common.AbstractWranglerHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * The REST APIs for managing and interacting with the standard schemas in the system.
+ */
+public class SchemaViewerHandler extends AbstractWranglerHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(SchemaViewerHandler.class);
+  private static final HealthcarePackageSchemaLoader SCHEMA_LOADER = new HealthcarePackageSchemaLoader();
+
+  class SchemaListingResponse {
+    private String id;
+    private SchemaFile.Standard standard;
+    private String version;
+
+    SchemaListingResponse(String id, SchemaFile.Standard standard, String version) {
+      this.id = id;
+      this.standard = standard;
+      this.version = version;
+    }
+  }
+
+  /**
+   * Populates the SchemaFileStore with the healthcare specific schema definitions.
+   * @param context the HTTP service runtime context
+   * @throws Exception
+   */
+  @Override
+  public void initialize(SystemHttpServiceContext context) throws Exception {
+    super.initialize(context);
+
+    try {
+      TransactionRunners.run(getContext(), ctx -> {
+        SchemaFileStore store = SchemaFileStore.get(ctx);
+        if (store.empty()) {
```

Comment:
this is subject to race conditions. For example, if half of the schemas were added to the store and then the service crashes, the store will never be fully populated. It's better to store a flag that indicates whether the store got initialized or not, perhaps by adding a new key to the ConfigStore